jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -690,6 +701,10 @@ ecma_op_function_construct_simple_or_external (ecma_object_t *func_obj_p, /**< F
                                          arguments_list_len),
                   ret_value);
 
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+  JERRY_CONTEXT (class_invoked_with_new) = false;
```

Comment:
Who sets this value to false if `ecma_op_function_call` throws an error? Using a global value is risky since it always finds way to escape :(

Reply:
This can be improved after #2414 lands.